ddtrace/propagation/http.py

```diff
@@ -106,14 +106,8 @@ def _extract_header_value(possible_header_names, headers, default=None):
 
 def _hex_id_to_dd_id(hex_id):
     # type: (str) -> int
-    """Helper to convert hex ids into Datadog compatible ints
-    If the id is > 64 bit and 128bit trace_id generation is disabled then truncate the trailing 64 bit.
-    "463ac35c9f6413ad48485a3953bb6124" -> "48485a3953bb6124" -> 5208512171318403364
-    Otherwise convert the full trace id into lower case hex values.
-    """
-    if config._128_bit_trace_id_enabled:
-        return int(hex_id, 16)
-    return int(hex_id[-16:], 16)
```

Comment:
So by no longer truncating it here, are we not keeping them as 128 bit trace IDs? I thought we have to ensure trace IDs sent to the agent are 64 bits.

Reply:
The w3c and b3 trace ids will be 128 bits within a process. This 128 bit trace id will be used in sampling decisions and to parent spans. However when the trace id leaves the process (ie: encoded then sent to the agent or in a distributed trace) then it will be truncated to 64bits. I updated the PR description with more details.

This change should be backwards will all agent versions.

Reply:
where is the truncating to 64 bits happening if not here? If I'm understanding this correctly, this change is taking out the responsibility of truncating to 64 bits out of the http propagation module?

Reply:
Yup truncation should only happen during encoding. It should never happen in the http module.